### PR TITLE
Avoid possible `InvalidOperationException` in `NewPSSessionOptionCommand.MaximumRedirection`

### PR DESCRIPTION
**Pull request by xtqqczze**
Specify a default value rather then allowing `_maximumRedirection.Value` to throw the exception.

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/commands/NewPSSessionOptionCommand.cs [hunk 1]
```diff
@@ -29,9 +29,8 @@ public sealed class NewPSSessionOptionCommand : PSCmdlet
         [Parameter]
         public int MaximumRedirection
         {
-            get { return _maximumRedirection.Value; }
-
-            set { _maximumRedirection = value; }
+            get => _maximumRedirection ?? -1;
+            set => _maximumRedirection = value;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/NewPSSessionOptionCommand.cs:33]: [From docs](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/new-pssessionoption?view=powershell-7.1):
> The default value is 5. A value of 0 (zero) prevents all redirection.

Is real default 0? /cc @PaulHigin

Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/commands/NewPSSessionOptionCommand.cs]: Default changed to `0`.

**Review by PaulHigin — CHANGES_REQUESTED** [review]
Actually, this doesn't look right to me.  The private field `_maximumRedirected` is different than what the getter returns, and it is used internally:

```csharp
            if (_maximumRedirection.HasValue)
            {
                result.MaximumConnectionRedirectionCount = this.MaximumRedirection;
            }
```

This code is weird, but I don't think we need to change it, unless it is to remove the private field.

**Comment by xtqqczze** [comment]
> Actually, this doesn't look right to me. The private field `_maximumRedirected` is different than what the getter returns, and it is used internally:
> 
> ```cs
>             if (_maximumRedirection.HasValue)
>             {
>                 result.MaximumConnectionRedirectionCount = this.MaximumRedirection;
>             }
> ```
> 
> This code is weird, but I don't think we need to change it, unless it is to remove the private field.

@PaulHigin I don't think we can remove the private field. `PSSessionOption.MaximumConnectionRedirectionCount` is initialized to `WSManConnectionInfo.defaultMaximumConnectionRedirectionCount` (which is `5`). So unless we make breaking API change by making `MaximumRedirection` nullable, I think we need to keep `_maximumRedirection`. 

The code is hard to understand, therefore unless you can recommend specific changes I think we should close this PR and accept the possible `InvalidOperationException`.

**Pull request closed without merging**